src/server/test/db/unitReadingsTests.js

```diff
@@ -0,0 +1,605 @@
+				{ start_timestamp: timestamp1 });
+			expect(meter_id).to.equal(meter.id);
+			expect(reading_rate).to.equal(100);
+		});
+
+		mocha.it('Daily readings when readings line up with hour for part of day', async () => {
+			await Reading.insertAll([
+				new Reading(meter.id, 100, timestamp1, timestamp2),
+				new Reading(meter.id, 200, timestamp2, timestamp3),
+				new Reading(meter.id, 300, timestamp3, timestamp4),
+				new Reading(meter.id, 400, timestamp4, timestamp5)
+			], conn);
+			await Reading.refreshDailyReadings(conn);
+			const { meter_id, reading_rate } = await conn.one(
+				'SELECT * FROM daily_readings_unit WHERE time_interval && tsrange(${start_timestamp}, ${end_timestamp});',
+				{ start_timestamp: timestamp1, end_timestamp: timestamp2 });
+			expect(meter_id).to.equal(meter.id);
+			expect(reading_rate).to.equal((100 + 200 + 300 + 400) / 4);
+		});
+
+		mocha.it('Daily readings when one reading half hour half hour before/after start of day', async () => {
+			const startOfDay = moment('2018-01-01');
+			const halfHourBefore = startOfDay.clone().subtract(30, 'minutes');
+			const halfHourAfter = startOfDay.clone().add(30, 'minutes');
+			await Reading.insertAll([
+				new Reading(meter.id, 100, halfHourBefore, halfHourAfter)
+			], conn);
+
+			await Reading.refreshDailyReadings(conn);
+			const rows = await conn.many('SELECT * FROM daily_readings_unit;');
+			expect(rows).to.have.length(2);
+			expect(rows[0].meter_id).to.equal(meter.id);
+			expect(rows[1].meter_id).to.equal(meter.id);
+
+			expect(rows[0].reading_rate).to.equal(100);
+			expect(rows[1].reading_rate).to.equal(100);
+		});
+
+		mocha.it('Daily readings with two readings of unequal length that overlap start and end of day', async () => {
+			const day1Start = moment('2018-01-01');
+			const day2Start = moment('2018-01-02');
+
+			await Reading.insertAll([
+				new Reading(meter.id, 100, day1Start.clone().subtract(1, 'hour'), day1Start.clone().add(1, 'hour')),
+				new Reading(meter.id, 400, day2Start.clone().subtract(2, 'hours'), day2Start.clone().add(2, 'hours'))
+			], conn);
+
+			// First reading: 100 kWh over 2 hours = 50 kW, 1 hour of overlap with the time interval
+			// Second reading: 400 kwH over 4 hours = 100 kW, 2 hours of overlap with the time interval
+
+			// Expected compressed reading:
+			// ((50 kW * 1 hr) + (100 kW * 2 hr)) / (1 hr + 2 hr)
+
+			await Reading.refreshDailyReadings(conn);
+
+			const { meter_id, reading_rate } = await conn.one('SELECT * FROM daily_readings_unit WHERE lower(time_interval) = ${start_timestamp};',
+				{ start_timestamp: day1Start });
+
+			expect(meter_id).to.equal(meter.id);
+			expect(reading_rate).to.be.closeTo(((50 * 1) + (100 * 2)) / (1 + 2), 0.0001);
+		});
+	});
+
+	mocha.describe('Check intervals', () => {
+		let meter, graphicUnitId, conn;
+		const startDate = '2020-01-01 00:00:00';
+		// Do 61 days which is what is needed.
+		const endDate = '2020-03-02 00:00:00';
```

Comment:
I found that startDate and endDate are not necessary. Do you think we should remove it?

Reply:
Agreed, thanks & done.